False positive for rule `comma` inside image literals

### New Issue Checklist

- [✔ ] Updated SwiftLint to the latest version
- [✔ ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

Having  `#imageLiteral(resourceName: "foo,bar,baz")` in your project will trigger a warning, and will be modified when using autocorrect

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint
Linting Swift files at paths
Linting 'ViewController.swift' (1/3)
Linting 'AppDelegate.swift' (2/3)
Linting 'LoadingViewController.swift' (3/3)
/Users/dkiszkiel/Desktop/TestApp/TestApp/AppDelegate.swift:14: warning: Line Length Violation: Line should be 120 characters or less: currently 145 characters (line_length)
/Users/dkiszkiel/Desktop/TestApp/TestApp/ViewController.swift:15:33: warning: Comma Spacing Violation: There should be no space before and one after any comma. (comma)
/Users/dkiszkiel/Desktop/TestApp/TestApp/ViewController.swift:15:37: warning: Comma Spacing Violation: There should be no space before and one after any comma. (comma)
Done linting! Found 3 violations, 0 serious in 3 files.
```

### Environment

* SwiftLint version 0.27.0
* Installation method used: homebrew
* running without config

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,7 +11,9 @@
 
 #### Bug Fixes
 
-* None.
+* Fix `comma` rule false positives on object literals (for example, images).  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#2345](https://github.com/realm/SwiftLint/pull/2345)
 
 ## 0.27.0: Heavy Duty
 

--- a/Rules.md
+++ b/Rules.md
@@ -1598,6 +1598,10 @@ func abc(
 
 ```
 
+```swift
+#imageLiteral(resourceName: "foo,bar,baz")
+```
+
 </details>
 <details>
 <summary>Triggering Examples</summary>

--- a/Source/SwiftLintFramework/Rules/Style/CommaRule.swift
+++ b/Source/SwiftLintFramework/Rules/Style/CommaRule.swift
@@ -17,7 +17,8 @@ public struct CommaRule: CorrectableRule, ConfigurationProviderRule, AutomaticTe
             "abc(a: \"string\", b: \"string\"",
             "enum a { case a, b, c }",
             "func abc(\n  a: String,  // comment\n  bcd: String // comment\n) {\n}\n",
-            "func abc(\n  a: String,\n  bcd: String\n) {\n}\n"
+            "func abc(\n  a: String,\n  bcd: String\n) {\n}\n",
+            "#imageLiteral(resourceName: \"foo,bar,baz\")"
         ],
         triggeringExamples: [
             "func abc(a: String↓ ,b: String) { }",
@@ -82,8 +83,8 @@ public struct CommaRule: CorrectableRule, ConfigurationProviderRule, AutomaticTe
         "\(mainPatternGroups)"      // Regexp will match if expression begins with comma
 
     private static let regularExpression = regex(pattern, options: [])
-    private static let excludingSyntaxKindsForFirstCapture = SyntaxKind.commentAndStringKinds
-    private static let excludingSyntaxKindsForSecondCapture = SyntaxKind.commentKinds
+    private static let excludingSyntaxKindsForFirstCapture = SyntaxKind.commentAndStringKinds.union([.objectLiteral])
+    private static let excludingSyntaxKindsForSecondCapture = SyntaxKind.commentKinds.union([.objectLiteral])
 
     private func violationRanges(in file: File) -> [NSRange] {
         let contents = file.contents

